.gitignore

```diff
@@ -16,6 +16,7 @@
 # production
 /build
 /public/umami.js
+/public/snippet.js
```

Comment:
I think this is getting way out of scope. It should be simple for the common user while still support your use case. I don't think it's common to use two of the same script on a page. I think by disabling the manual collection and having you call the methods directly with a given website uuid it should be enough,

Reply:
This was not to only support my use case of multiple trackers. This is just the best way of implementing tracker for end-users.

Consider this scenario:
1) I want to set up Umami tracker (or Segment.io) tracker. Ideally, it should be placed in <head> but I don't want it to be blocking rendering. So I set the script tag as `async` and still place it in the head.
2) But now I have a problem, I want to call a custom event when the user scrolls down below the fold. The `window.umami` can be undefined because the script can take some time to load. 
3) The way to handle it (which I shamelessly copied from Segment.io tracker) is to have a tracking snippet contain small logic, to save all calls of `window.umami`, and to have the tracker replay those.

Now, with current logic, you can place tracker code in <head> with async, not blocking loading, but at the same time you can use `window.umami.event` immediately, even if the tracker takes 20secs to load. Once it loads, all the events will be logged then.

Reply:
Example from Google Analytics:
```
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-XXXXXX-XX"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-3585260-17');
</script>
```

Reply:
Yes, I understand how it works and agree it is a more complete solution. But my main goal is to keep the tracking as simple and easy to use as possible. A lot of people already have the original tracker installed and to have them change to the snippet is very disruptive.

I think it would be better to have this as a guide on the website as an advanced use case and provide steps on how to set it up with snippet code. But I really don't like the snippet as the default behavior. Most people will not even use events.

Otherwise the other changes look good. Having a `window.umami` object that you can call manually is very useful.

Reply:
Yes, I agree with all the points. I will change default snippet back to the simpler one and make sure the tracker is fully backwards compatible. 👌